pkg/oc/clusterup/run_self_hosted.go

```diff
@@ -206,177 +101,126 @@ func (c *ClusterUpConfig) StartSelfHosted(out io.Writer) error {
-		kubeDNSConfigDir:             filepath.Join(c.BaseDir, kubelet.KubeDNSDirName),
-		podManifestDir:               filepath.Join(c.BaseDir, kubelet.PodManifestDirName),
-	}
-
-	originalMasterConfigDir := configs.masterConfigDir
-	originalNodeConfigDir := configs.nodeConfigDir
-
-	if _, err := os.Stat(configs.masterConfigDir); os.IsNotExist(err) {
-		_, err = c.makeMasterConfig()
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if _, err := os.Stat(configs.openshiftAPIServerConfigDir); os.IsNotExist(err) {
-		_, err = c.makeOpenShiftAPIServerConfig(originalMasterConfigDir)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	if _, err := os.Stat(configs.openshiftControllerConfigDir); os.IsNotExist(err) {
-		_, err = c.makeOpenShiftControllerConfig(originalMasterConfigDir)
-		if err != nil {
-			return nil, err
-		}
+		podManifestDir: filepath.Join(c.BaseDir, kubelet.PodManifestDirName),
+		nodeConfigDir:  filepath.Join(c.BaseDir, kubelet.NodeConfigDirName),
+		assetsDir:      filepath.Join(c.BaseDir, "bootkube"),
+		kubernetesDir:  filepath.Join(c.BaseDir, "kubernetes"),
 	}
 
-	if _, err := os.Stat(configs.nodeConfigDir); os.IsNotExist(err) {
-		_, err = c.makeNodeConfig(configs.masterConfigDir)
-		if err != nil {
+	if _, err := os.Stat(configs.assetsDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(configs.assetsDir, 0755); err != nil {
 			return nil, err
 		}
 	}
-
-	if _, err := os.Stat(configs.kubeDNSConfigDir); os.IsNotExist(err) {
-		_, err = c.makeKubeDNSConfig(originalNodeConfigDir)
-		if err != nil {
+	if _, err := os.Stat(configs.kubernetesDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(configs.kubernetesDir, 0755); err != nil {
 			return nil, err
 		}
-
 	}
-
 	if _, err := os.Stat(configs.podManifestDir); os.IsNotExist(err) {
 		if err := os.MkdirAll(configs.podManifestDir, 0755); err != nil {
 			return nil, err
 		}
-
 	}
 
-	substitutions := map[string]string{
-		"/path/to/master/config-dir":              configs.masterConfigDir,
-		"/path/to/openshift-apiserver/config-dir": configs.openshiftAPIServerConfigDir,
-		"ETCD_VOLUME":                             "emptyDir:\n",
-		"OPENSHIFT_PULL_POLICY":                   c.pullPolicy,
+	glog.V(2).Infof("Running bootkube render %q", configs.podManifestDir)
+	bk := bootkube.BootkubeRunConfig{
+		BootkubeImage:        "quay.io/coreos/bootkube:v0.13.0",
+		StaticPodManifestDir: configs.podManifestDir,
+		AssetsDir:            configs.assetsDir,
+		ContainerBinds:       []string{},
 	}
 
-	if len(c.HostDataDir) > 0 {
-		substitutions["ETCD_VOLUME"] = `hostPath:
-      path: ` + c.HostDataDir + "\n"
-	}
+	hostIP, err := c.determineIP()
+	if err != nil {
+		return nil, err
+	}
+	if _, err := bk.RunRender(c.GetDockerClient(), hostIP); err != nil {
+		return nil, err
+	}
+
+	// TODO: These should go away if we make the bootkube upstream more extendable
+	if err := bk.Substitute("bootstrap-manifests/bootstrap-apiserver.yaml", map[string]string{
```

Comment:
@deads2k @sttts these are ugly, but I would rather have them here than fixing this in bootkube (also bootkube have these built into the version, so it won't break us unless we bump the bootkube image version)